cluster-location-metadata/README.md

```diff
@@ -0,0 +1,332 @@
+
+With Cluster API on cloud providers we usually have location indicators in cluster resources, in various places. Each cloud provider uses their own identifier system, however they all make use of a string identifier. The proposed solution in this RFC would help simplify and unify location lookup for clusters by clients throughout cloud providers.
+
+In on-premises installations, no location information is currently available in Kubernetes resources. This RFC attempts to change that.
+
+### Key assumptions
+
+This RFC assumes that each cluster exists in one location only, and nodes of the same cluster are not distributed geographically.
+
+### Decision maker
+
+- Team Honeybadger as driver
+- KaaS teams, especially Rocket
+
+### Deadline
+
+None
+
+### Who is affected / stakeholders
+
+- Customers
+- Giant Swarm engineers
+
+### Preferred solution
+
+The solution proposed here consists of two parts:
+
+- Setting consistent metadata on clusters
+- Providing a default location ID system for on-prem customers
+
+#### Consistent metadata on clusters
+
+We should set the [`topology.kubernetes.io/region`](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesioregion) label on the main cluster resource of all clusters. Currently (as of February 2025) this resource is of kind `Cluster`, group=`cluster.x-k8s.io/v1beta1`.
+
+The `topology.kubernetes.io/region` label is documented as w well-known label, set by Kubernetes on `Node` and `PersistentVolume` resources.
+
+For clusters running in cloud providers, the value of the label should be the region identifier. E. g. `eu-west-1` for the AWS region in EU/Dublin.
+
+For clusters on premises, the value should follow a system that we establish for this purpose. The system may differ between customers, to allow fulfilling of customer-specific requirements.
+
+Regardless of the environment (cloud/on-prem), the value of the `topology.kubernetes.io/region` label should not include any whitespace and consist of the lowercase letters, numbers, and dash only (`[a-z0-9-]`).
+
+#### Default location ID system for on-prem
+
+The metadata system described above allows to specify the location of a cluster as a single string.
+
+For our on-premises customers, we want to provide a default system to inidicate their cluster's locations. This default system should be
+
+- easy to use and to understand
+- hierarchical
+- adaptable/customizable
+
+It should be up to the customer to decide if they want to deviate from the system and populate the label mentioned above with whatever string they like. For example, let's just assume a customer has a numeric system to enumerate their facilities. In that case, they should be able to use that identifier instead of our default system.
+
+For our default system, here is a synopsis of the format:
+
+```
+<CONTINENT_CODE>-<COUNTRY_CODE>[-<SUBDIVISION_CODE>[-<CITY_NAME>]]
```

Comment:
This looks fine to me. I agree with Xav that there could be ambiguity about the meaning of the optional parts, but if we need to rely on that for some reason, that use case probably warrants its own proper support.

Requiring the country to be indicated for all clusters will help with compliance stories. Borders might change, but the label can follow.

I'd offer a preemptive suggestion that there should probably be a required, incrementing number for each location path. The first time a customer has two locations in a given city, we'll have `ab-cd-ef-gh` (which is the first location) and `ab-cd-ef-gh-1` which is, counterintuitively, the second location. It' easy now to start with `-1` and increment.

Reply:
Bumping the numbering suggestion here -- does anyone have thoughts for/against?

Reply:
The only downside I see is that the number isn't self-explanatory. But that's what labels are for.

When writing the proposal, I had in mind that the system would have to grow with the customer's needs. To use your example scenario, given `eu-fr-cityname`, they would have to change that to `eu-fr-cityname-foo` and add ``eu-fr-cityname-bar`. But of course it would be less complicated if changing wasn't necessary.

Reply:
It makes sense to me to have a trailing numeric ID, it certainly won't hurt at least.
In that case do we need the subdivision code? It's starting to be quite a long string.